File: Programmers/Lv.1/최소직사각형.py
def solution(sizes):
    answer = 0
    max_x = 0
    max_y = 0
    
    for i in range(len(sizes)):
        if sizes[i][0] < sizes[i][1]:
            temp = sizes[i][0]
            sizes[i][0] = sizes[i][1]
            sizes[i][1] = temp
        
        max_x = max(sizes[i][0], max_x)
        max_y = max(sizes[i][1], max_y)
    
    answer = max_x * max_y
    
    return answer

# def solution(sizes):
#     return max(max(x) for x in sizes) * max(min(x) for x in sizes)

File: Programmers/Lv.1.Kotlin/최소직사각형.kt
import kotlin.math.*

class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        var arr = sizes.mapIndexed { indexArr, items ->
            if (items[0] < items[1]) {
                var temp = items[0]
                items[0] = items[1]
                items[1] = temp
            }
        }.toTypedArray()
        
        var maxX = 0
        var maxY = 0
        
        for (i in 0..sizes.size - 1) {
            maxX = max(sizes[i][0], maxX)
            maxY = max(sizes[i][1], maxY)
        }
        
        return maxX * maxY
    }

    // fun solution(sizes: Array<IntArray>): Int {
    //     var a = 0
    //     var b = 0

    //     for (array in sizes) {
    //         array.sort()
    //         if (array.isNotEmpty()) {
    //             a = array.first().coerceAtLeast(a)
    //             b = array.last().coerceAtLeast(b)
    //         }
    //     }
    //     return a * b
    // }
}